projects/cdk/schematics/ng-update/steps/show-warnings.ts

```diff
@@ -12,12 +14,27 @@ function showWarning(
 ): void {
     const references = getNamedImportReferences(name, moduleSpecifier);
 
-    references.forEach(reference => {
-        const file = reference.getSourceFile().getFilePath();
-        const {line, column} = reference
-            .getSourceFile()
-            .getLineAndColumnAtPos(reference.getStart());
+    const referencesMeta = references.map(ref => {
+        const sourceFile = ref.getSourceFile();
+
+        return {
+            sourceFile,
+            filePath: sourceFile.getFilePath().toString(),
+            isImport: Node.isImportSpecifier(ref.getParent()),
+            startLinePos: ref.getStartLinePos(),
+        } as const;
+    });
+
+    /**
+     * We have to twice iterate array with refs because otherwise we get error:
+     * > Attempted to get information from a node that was removed or forgotten.
+     * See this {@link https://ts-morph.com/manipulation/#strongwarningstrong warning}
```

Comment:
nit: Node has method `wasForgotten` (I don't know can it help in your case, but anyway it can be useful)